src/Microsoft.TestPlatform.Common/Filtering/FilterExpression.cs

```diff
@@ -127,30 +127,59 @@ private static void ProcessOperator(Stack<FilterExpression> filterStack, Operato
             properties = Enumerable.Empty<string>();
         }
 
-        bool valid = false;
-        if (_condition != null)
+        FilterExpression? current = this;
+        Stack<FilterExpression> filterStack = new();
+        Stack<string[]?> result = new();
+        do
         {
-            valid = _condition.ValidForProperties(properties, propertyProvider);
-            if (!valid)
+            //Push root's right child and then root to stack then Set root as root's left child
```

Comment:
```suggestion
            // Push root's right child and then root to stack then Set root as root's left child.
```

Reply:
I believe that entire comment is unnecessary, and more complicated than the code :)